src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -1098,6 +1108,34 @@ namespace winrt::TerminalApp::implementation
         return _zoomedPane != nullptr;
     }
 
+    // Method Description:
+    // - Toggle read-only mode on the active pane
+    void TerminalTab::TogglePaneReadOnly()
+    {
+        auto control = GetActiveTerminalControl();
+        if (control)
+        {
+            control.ToggleReadOnly();
+        }
+    }
+
+    // Method Description:
+    // - Calculates if the tab is read-only.
+    // The tab is considered read-only if one of the panes is read-only.
+    // If after the calculation the tab is read-only we hide the close button on the tab view item
+    // Currently this is the only indicator that the tab is read-only.

```

Comment:
```suggestion
```
Is there a follow-up to add a lock icon to the tab?
I'd remove this statement since that'll change soon, and wouldn't want to forget to update it. Teeny nit though.

Reply:
So regarding this I am not sure if there will be a follow up - the lock icon appears only when the locked **pane** is focused.
But when the locked pane is not focused we want to somehow indicate that another pane in this tab is locked.

So right now we simply hide the close button. Alternatively, we could also add another glyph (instead of the lock) to indicate that inactive pane is locked.

WDYT?

Reply:
>  the lock icon appears only when the locked pane is focused... So right now we simply hide the close button

I think I'm okay with that UX personally. There's not usually _that many_ panes in a tab that tracking down the one that's read only would be that challenging...